Include/internal/pycore_freelist.h

```diff
@@ -147,6 +156,39 @@ extern void _PyAsyncGen_ClearFreeLists(struct _Py_object_freelists *freelists, i
 extern void _PyContext_ClearFreeList(struct _Py_object_freelists *freelists, int is_finalization);
 extern void _PyObjectStackChunk_ClearFreeList(struct _Py_object_freelists *freelists, int is_finalization);
 
+// Keep in sync with _asynciomodule.c !
+typedef struct futureiterobject_dummy {
```

Comment:
As mentioned by Sam we should try to find a way to avoid this.

Reply:
If we got with https://github.com/python/cpython/pull/121934, then that will make the asyncio freelist implementation a lot easier and will avoid a lot of these warts.

Maybe, in the meantime, we should just disable the freelist in the free-threaded build? That would be a much smaller change and wouldn't introduce duplicate types.

Reply:
We could backport the disabling of freelist in the free-threaded build to 3.13 to fix the race conditions.